File: 백준/Gold/1068. 트리/트리.py
n = int(input())
parent = list(map(int, input().split()))
delNode = int(input())


def DFS(node):
    parent[node] = -2
    for i in range(n):
        if parent[i] == node:
            DFS(i)


DFS(delNode)

ans = 0
for i in range(n):
    if i not in parent and parent[i] != -2:
        ans += 1

print(ans)

File: 백준/Gold/1068. 트리/트리.java
import java.io.*;
import java.util.*;

public class Main {

    static int N,toRemove;
    static int[] parent;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());
        parent = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        toRemove = Integer.parseInt(br.readLine());

        DFS(toRemove);

        int answer = 0;
        for (int i = 0; i < N; i++) {
            boolean status = false;
            for (int p : parent) {
                if (p == i) status = true;
            }
            if (!status && parent[i] != -2) {
                answer++;
            }

        }

        System.out.println(answer);
    }

    public static void DFS(int node) {
        parent[node] = -2;
        for (int i = 0; i < N; i++) {
            if (parent[i] == node) DFS(i);
        }
    }

}